client/src/www/views/root_view/add_access_key_dialog/index.ts

```diff
@@ -105,13 +112,22 @@ export class AddAccessKeyDialog extends LitElement {
         </md-text-button>
         <md-filled-button
           @click=${this.handleConfirm}
-          ?disabled=${!this.accessKey || !this.isValidAccessKey(this.accessKey)}
+          ?disabled=${!this.accessKey || !this.isValidAccessKey}
           >${this.localize('confirm')}</md-filled-button
         >
       </fieldset>
     </md-dialog>`;
   }
 
+  private updateIsValidAccessKey(
+    accessKey: string,
+    validate: (accessKey: string) => Promise<boolean>
+  ) {
+    validate(accessKey).then(result => {
```

Comment:
Make sure the buttons are disabled during validation? User might paste a valid key first (so the buttons are enabled), and then replace it with another invalid key (the buttons are still enabled, until the validation finishes after some time).

```suggestion
    this.isValidAccessKey = false;
    validate(accessKey).then(result => {
```